Src/HALAL/Services/Communication/SPI/SPI.cpp

```diff
@@ -47,42 +54,49 @@ bool SPI::transmit(uint8_t id, uint8_t data){
 }
 
 bool SPI::transmit(uint8_t id, span<uint8_t> data) {
-    if (!SPI::registered_spi.contains(id)) {
-        ErrorHandler("No SPI registered with id %u", id);
-    	return false;
-    }
-
-    SPI::Instance* spi = SPI::registered_spi[id];
+	 if (!SPI::registered_spi.contains(id)){
+		ErrorHandler("No SPI registered with id %u", id);
+		return false;
+	 }
 
-    turn_off_chip_select(spi);
-    if (HAL_SPI_Transmit(spi->hspi, data.data(), data.size(), 10) != HAL_OK){
-    	turn_on_chip_select(spi);
-        ErrorHandler("Error during transmission in %s", spi->name.c_str());
-    	return false;
-    }
-	turn_on_chip_select(spi);
+	SPI::Instance* spi = SPI::registered_spi[id];
 
-    return true;
+	 HAL_StatusTypeDef errorcode = HAL_SPI_Transmit_DMA(spi->hspi, data.data(), data.size());
+	 switch(errorcode){
+	 	 case HAL_OK:
+		 	 return true;
+		 break;
+	 	 case HAL_BUSY:
+	 		 return false;
+	 	 break;
+	 	 default:
+	 		 ErrorHandler("Error while transmiting and receiving with spi DMA. Errorcode %u",(uint8_t)errorcode);
+	 		 return false;
+	 	 break;
+	 }
```

Comment:
no need to break after returning, break is normally used in switch statements because by default swith statements fallthrough